Xamarin.Forms.Controls/Xamarin.Forms.Controls.csproj

```diff
@@ -30,7 +30,9 @@
     <ProjectReference Include="..\Xamarin.Forms.Maps\Xamarin.Forms.Maps.csproj" />
     <ProjectReference Include="..\Xamarin.Forms.Xaml\Xamarin.Forms.Xaml.csproj" />
   </ItemGroup>
-  <Import Project="..\.nuspec\Xamarin.Forms.targets" />
+  
+  <Import Project="..\.nuspec\Xamarin.Forms.targets" Condition="'$(BuildingInsideVisualStudio)' == 'true' AND Exists('..\.nuspec\Xamarin.Forms.Build.Tasks.dll')" />
+  <Import Project="..\.nuspec\Xamarin.Forms.targets" Condition="'$(BuildingInsideVisualStudio)' != 'true'" />
```

Comment:
I don't get how the `.targets` will be included if the `Build.Tasks` hasn't been built yet. This csproj has a dependency on the `Build.Tasks` (iirc) and will built it, but that won't retrigger the parsing of the csproj and the reevaluation of the condition. Or will it ?

Reply:
To be honest, I'm not sure if VS is re-evaluating the condition or if VS just never considers XF.targets to be "imported". But MSBuild doesn't evaluate that condition until after the dependencies have been built, so the build works fine even if Build.Tasks doesn't exist when it starts.

But if VS considers the project load to have "failed" initially, it won't build the project and it won't change that "failed" status until the project is re-loaded. If it considers it "successful", it will happily hand it off to MSBuild to do the rest.